### ISSUE TITLE
Incorrect implementation of SGD momentum

### ISSUE THREAD

**Issue by fostiropoulos** [pytorch/pytorch#44917]
## 🐛 Bug

Pytorch documentation: https://pytorch.org/docs/stable/optim.html#torch.optim.SGD
"This is in contrast to Sutskever et. al. and other frameworks which employ an update of the form"
![image](https://user-images.githubusercontent.com/4337024/93537302-022c5f00-f900-11ea-9a98-fa83a7b90654.png)

SGD: http://proceedings.mlr.press/v28/sutskever13.pdf
![image](https://user-images.githubusercontent.com/4337024/93537287-f9d42400-f8ff-11ea-9535-42d1eafb92f1.png)

As an implication both the adapted and the interpreted version of SGD have negative momentum

**Comment by fostiropoulos** [comment]
I am not sure if this error also reflects in the implementation of SGD.

**Comment by fostiropoulos** [comment]
The error carries out in the implementation 

https://pytorch.org/docs/stable/_modules/torch/optim/sgd.html#SGD

```python
            for p in group['params']:
                if p.grad is None:
                    continue
                d_p = p.grad
                if weight_decay != 0:
                    d_p = d_p.add(p, alpha=weight_decay)
                if momentum != 0:
                    param_state = self.state[p]
                    if 'momentum_buffer' not in param_state:
                        buf = param_state['momentum_buffer'] = torch.clone(d_p).detach()
                    else:
                        buf = param_state['momentum_buffer']
                        buf.mul_(momentum).add_(d_p, alpha=1 - dampening)
                    if nesterov:
                        d_p = d_p.add(buf, alpha=momentum)
                    else:
                        d_p = buf

                p.add_(d_p, alpha=-group['lr'])
```

### Analysis of differences
#### Problem 1:

Wrong interpretation of SGD algorithm by [Sutskever et. al. (2013)](http://proceedings.mlr.press/v28/sutskever13.pdf)

```
                v_{t+1} & = \mu * v_{t} + \text{lr} * g_{t+1}, \\
                p_{t+1} & = p_{t} - v_{t+1}.
```

Should instead be (notice the signs): 

```
                v_{t+1} & = \mu * v_{t} - \text{lr} * g_{t+1}, \\
                p_{t+1} & = p_{t} + v_{t+1}.
```

#### Problem 2:

Momentum as it currently is implemented results in a negative value. For example:

                v_{t+1} & = \mu * v_{t} + g_{t+1}, \\
                p_{t+1} & = p_{t} - \text{lr} * v_{t+1},

is equivalent to 

                p_{t+1} & = p_{t} - \text{lr} *  \mu * v_{t} -  \text{lr} * g_{t+1}

In contrast to the update rule by Sutkever:

                p_{t+1} & = p_{t} + \mu * v_{t} - \text{lr} * g_{t+1},

So the difference is not only the momentum is scaled depending on the learning rate when compared to vanilla SGD, but also multiplied by -1. 

### Resolution? 

If this is intended behavior should be better documented and at least **Problem 1** resolved.
